# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/2d/Filters.h
# Commit: 3c9634352cd6
# Full Hash: 3c9634352cd66e1debc3beeedd53130d16474db2
# Author: Jeff Muizelaar <jrmuizel@gmail.com>
# Date: 2019-01-28 04:51:03
# Regressor Bug: 1522021
# File Overlap Count: 1
# Description:
#   Bug 1522021. Propagate input size to CreateSimilarDrawTarget for filters. r=mstange
#   
#   This lets us avoid drawing the complete input for ever tile when
#   drawing filters into a tile.
#   
# ==============================================================================

diff -r 747303cab27a -r 3c9634352cd6 gfx/2d/Filters.h
--- a/gfx/2d/Filters.h	Sun Jan 27 22:06:38 2019 +0000
+++ b/gfx/2d/Filters.h	Sun Jan 27 23:36:04 2019 +0000
@@ -425,6 +425,14 @@
     MOZ_CRASH("GFX: FilterNode");
   }
 
+  /** Maps a rectangle in filter space back to a rectangle in the space of
+   * aSourceNode's first input. aSourceNode should not have an input
+   * assigned when calling this function. */
+  virtual IntRect MapRectToSource(const IntRect &aRect, const IntRect &aMax,
+                                  FilterNode *aSourceNode) {
+    return aMax;
+  }
+
  protected:
   friend class Factory;
 